subsys/net/lib/lwm2m/lwm2m_shell.c

```diff
@@ -625,6 +626,87 @@ static int cmd_cache(const struct shell *sh, size_t argc, char **argv)
 #endif
 }
 
+static void shell_print_attr(const struct shell *sh, void *ref)
+{
+	struct lwm2m_attr *attr = NULL;
+	bool found;
+	for (uint8_t type = 0; type < NR_LWM2M_ATTR; type++) {
+		found = false;
+		while ((attr = lwm2m_engine_get_next_attr(ref, attr)) != NULL) {
+			if (attr->type == type) {
+				found = true;
+				break;
+			}
+		}
+		if (found) {
+			switch (type) {
+			case LWM2M_ATTR_PMIN:
+			case LWM2M_ATTR_PMAX:
+				shell_fprintf(sh, SHELL_NORMAL, "%10u", attr->int_val);
+				break;
+			case LWM2M_ATTR_GT:
+			case LWM2M_ATTR_LT:
+			case LWM2M_ATTR_STEP:
+				shell_fprintf(sh, SHELL_NORMAL, "%10f", attr->float_val);
+				break;
+			}
+		} else {
+			shell_fprintf(sh, SHELL_NORMAL, "%10s", "");
+		}
+	}
+}
+
+static int cmd_observations(const struct shell *sh, size_t argc, char **argv)
+{
+	char buf[LWM2M_MAX_PATH_STR_SIZE];
+	struct lwm2m_obj_path_list *o_p;
+	struct observe_node *obs;
+	uint32_t i = 0, path_i;
+	struct lwm2m_ctx *ctx;
+	void *ref;
+	int ret;
+
+	ctx = lwm2m_rd_client_ctx();
+	if (ctx == NULL) {
+		shell_error(sh, "no lwm2m context yet\n");
+		return -ENOEXEC;
+	}
+
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	shell_fprintf(sh, SHELL_INFO, " # %10s%10s", "composite", "path");
+	for (i = 0; i < NR_LWM2M_ATTR; i++) {
+		shell_fprintf(sh, SHELL_INFO, "%10s", lwm2m_attr_to_str(i));
+	}
+	shell_print(sh, "");
+
+	lwm2m_registry_lock();
+	i = 0;
+	SYS_SLIST_FOR_EACH_CONTAINER(&ctx->observer, obs, node) {
+		shell_fprintf(sh, SHELL_NORMAL, "%2u %10c", i, obs->composite ? 'y' : 'n');
+		path_i = 0;
+		SYS_SLIST_FOR_EACH_CONTAINER(&obs->path_list, o_p, node) {
+			if (path_i > 0) {
+				shell_fprintf(sh, SHELL_NORMAL, "%13s", "");	
+			}
+			shell_fprintf(sh, SHELL_NORMAL, "%10s", lwm2m_path_log_buf(buf, &o_p->path));
```

Comment:
10 characters is not enough for paths like `65535/65535/65535`
Maybe 18?

Also, maybe pad to left, so `%-18s`

Reply:
I increased the width to 18 but I found the right alignment looked a bit better:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/131832/f37891c5-ff40-4261-be86-0eee75e2cd86)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/131832/463cc751-b00b-4cd6-becc-771f23da64b3)